### PR TITLE
feat(gatsby-cli): add GATSBY_EXPERIMENTAL_GATSBY_NEW_FLOW flag to use create-gatsby when no options are provided

### PR DESCRIPTION
**Pull request by gillkyle**
## Description

With the new `create-gatsby` package that doesn't use flags and provides an interactive new experience, this removes the old behavior that asked for a developer to choose a starter first when they used the `gatsby new` command without any flags or options passed in. The `create-gatsby` questions are more suited around use cases, instead of having to research what starter you'd want.

This also adds an integration test and changes the docs that mentioned the old interactive behavior.

_Note: the `create-gatsby` package at v0.0.0-6 still has a testing banner on it, so maybe we want to bump the package before we merge this? For that reason I'm making this a draft pull request to start._

image

### Documentation

I updated the `gatsby-cli` docs to mention this flow instead.

## Related Issues

Related to #27703 (once it's merged)

### CODE REVIEW

**Comment by ascorbic** [comment]
Looking good. It should have the testing banner as long as it's still in alpha. This PR can't come into master until create-gatsby is GA, because it changes the default behaviour of `gatsby new`. A feature flag is one possibility

**Comment by mxstbr** [comment]
We should add a feature flag for now so we can ship it and make sure it works as expected! `GATSBY_EXPERIMENTAL_GATSBY_NEW_FLOW=true`?

File: integration-tests/gatsby-cli/__tests__/new.js [hunk 1]
```diff
@@ -78,4 +78,10 @@ describe(`gatsby new`, () => {
 
     expect(code).toBe(1)
   })
+
+  it(`an empty gatsby new (without args) runs create-gatsby`, () => {
```
Comment (mfrachet) [inline integration-tests/gatsby-cli/__tests__/new.js:82]: Very very nitpicky: the test name is not a "readable" sentance. The "it" at the begining aims to start a sentance, something like `it runs create-gatsby when no arguments are provided to gatsby new`

Comment (gillkyle) [reply integration-tests/gatsby-cli/__tests__/new.js:82]: I love your nitpicks, thanks Marvin 🙂

**Comment by gillkyle** [comment]
Thanks for the suggestions 🙂  I just updated to add a `GATSBY_EXPERIMENTAL_GATSBY_NEW_FLOW` flag.

| Screenshot |
| --- |
|  image |

File: packages/gatsby-cli/src/create-cli.ts [hunk 2]
```diff
@@ -502,7 +503,15 @@ export const createCli = (argv: Array<string>): yargs.Arguments => {
         const starterStr = starter ? String(starter) : undefined
         const rootPathStr = rootPath ? String(rootPath) : undefined
 
-        await initStarter(starterStr, rootPathStr)
+        if (
+          process.env.GATSBY_EXPERIMENTAL_GATSBY_NEW_FLOW &&
+          !starterStr &&
+          !rootPathStr
+        ) {
```
Comment (mfrachet) [inline packages/gatsby-cli/src/create-cli.ts:510]: Nitpicky again:

Would you mind adding an explicit comment on why this condition is here? 

`We only run the interactive CLI when there are no arguments in gatsby new`

Or storing the result of the condition in an intermediate variable that is self explanatory? For clarity reasons. I know test is covering these scenario, but personally, I would love to read the code and get what's going on directly :D

**Review by ascorbic — APPROVED** [review]
This looks great, and subject to Marvin's suggestions I'm happy to see this go in

**Review by mxstbr — APPROVED** [review]
Love it! :100:

**Review by mfrachet — DISMISSED** [review]
go for it! 🚢

**Comment by gillkyle** [comment]
I really don't think this affects the CI tests so I'm going to go ahead and merge this since the code is behind a flag anyway and our new release process means it won't even be very publicly facing for a while (even being a Friday afternoon, YOLO) 😬

**Pull request merged**